Allow case-insensitive comparison for Equals as well just like Contains

wahmedswl opened the pull request:
fixes #78

wahmedswl committed:
Case insensitive comparison for Equals just like Contains

enisn commented:
Thanks for you effort, I'll merge it after creating at least a test for this case

enisn commented on src/AutoFilterer/Attributes/ToLowerEqualsComparisonAttribute.cs:
```diff
@@ -0,0 +1,30 @@
+
+        var toLowerMethod = typeof(string).GetMethod(nameof(string.ToLower), types: new Type[0]);
+
+        var comparison = Expression.Equal(
+                    Expression.Call(
+                        method: equalsMethod,
+                        instance: Expression.Call(method: toLowerMethod, instance: Expression.Property(context.ExpressionBody, context.TargetProperty.Name)
+                            ),
+                        arguments: new[] { Expression.Call(method: toLowerMethod, instance: context.FilterPropertyExpression) }),
+                    Expression.Constant(true));
```
No, it's not working like that. We need to use Expression in this case. Ignore this feedback

enisn commented on src/AutoFilterer/Attributes/ToLowerEqualsComparisonAttribute.cs:
```diff
@@ -0,0 +1,30 @@
+    {
+        var equalsMethod = typeof(string).GetMethod(nameof(string.Equals), types: new[] { typeof(string) });
+
+        var toLowerMethod = typeof(string).GetMethod(nameof(string.ToLower), types: new Type[0]);
```
That might be ok for some cases and not good for others. Maybe it can be configurable at some point. But existing filers lready has this issue. I'll consider this trade-off later

enisn committed:
Add tests for ToLowerEqualsComparisonAttribute

Introduces unit tests for the ToLowerEqualsComparisonAttribute, covering case insensitivity, empty and null values, special and unicode characters, whitespace, no matches, and exact match scenarios for the Book.Title property.
